src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,963 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+Option Explicit On
+Option Strict On
+
+Imports Microsoft.VisualBasic.FileIO
+Imports System.Collections.ObjectModel
+Imports System.Runtime.CompilerServices
+Imports System.Runtime.InteropServices
+Imports System.Text
+Imports Xunit
+' Do not Imports System.IO
+Namespace Microsoft.VisualBasic.Tests.VB
+    Public NotInheritable Class FileIOTests
+        Inherits IO.FileCleanupTestBase
+
+        ReadOnly DestData() As Char = {"x"c, "X"c, "y"c}
+
+        ReadOnly SourceData() As Char = {"a"c, "A"c, "b"c}
+
+        Sub New()
+        End Sub
+
+        Public Shared ReadOnly Property ManualTestsEnabled() As Boolean
+            Get
+                Return Not String.IsNullOrEmpty(Environment.GetEnvironmentVariable("MANUAL_TESTS"))
+            End Get
+        End Property
+
+        ''' <summary>
+        ''' All "Public" tests are Named for the FileIO function they test followed by _ParameterName for each Parameter and if there are options
+        ''' they are separated into additional test and the Option Value is the last  part of the name.
+        ''' For example CopyDirectory_SourceDirectoryName_DestinationDirectoryName_OverwriteFalse tests CopyDirectory with 3 arguments
+        ''' SourceDirectoryName, DestinationDirectoryName and Overwrite and the value of Overwrite being tested is False
+        ''' </summary>
+        Private Shared Function CreateTestFile(TestBase As FileIOTests, ByVal TestData() As Char, <CallerMemberName> Optional memberName As String = Nothing, <CallerLineNumber> Optional lineNumber As Integer = 0) As String
```

Comment:
`ByVal` can be removed from all parameters.

Reply:
Des this include those in the Reference source or just in routines I wrote? I will commit the later in a minute.

Reply:
I meant just in the methods you've written.